### PR TITLE
fix: remove superfluous words and updated model

### LINKED ISSUE #1: Remove markdown wrapping
PR descriptions already support markdown, so wrapping the description with backticks is not required.

Please:

- Remove the ```md code block wrapper
- Remove superfluous text after the PR changes list.  We don't need the model's opinion about PRs in general.

### LINKED ISSUE #2: Change models to support a larger context window
We should be using better models.  Let's try to implement GPT-4o

### PR DESCRIPTION
**Pull request by hhw67865**
- remove additional comments after the changes list.
- changed model to GPT-4o
- removed ```md code block

### CODE REVIEW

**Review by LukeZekes — APPROVED** [review]
Changes look good. I'm assuming the goal is to go straight from running the action to submitting the PR, without needing to greatly modify the generated text.

When running I did notice that
```
  template_markdown = if @pull_request_template != ""
        File.read("templates/#{@pull_request_template}")
      else
        "Give a description of the changes in a list of this PR. Use markdown, but do not enclose any part of the answer in backticks.  Do not add any advice or final thoughts."
      end
```
throws an error if `@pull_request_template` is its default value of `nil`. I know that's not really the scope of this PR though.

**Comment by hhw67865** [comment]
Good catch. I changed the if to handle the default nil value.

File: lib/pull_request_description_service.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def run
     # @return [String] the generated PR description
     def generate_prompt
       # If the pull_request_template.md file is set, read its contents
-      template_markdown = if @pull_request_template != ""
+      template_markdown = if @pull_request_template && !@pull_request_template.empty?
```
Comment (mostlydev) [inline lib/pull_request_description_service.rb:37]: This can simply be `@pull_request_template&.present?`

Comment (hhw67865) [reply lib/pull_request_description_service.rb:37]: I wanted to but  `.present?` is only a rails method.

**Comment by mostlydev** [comment]
OH, interesting.  I thought it was newer ruby.  Never mind then

**Pull request merged**